File: origins/flex.go
package renderers

type Flex struct {
	*BaseRenderer
}

func NewFlex() *Flex {
	f := &Flex{
		BaseRenderer: NewBaseRenderer(),
	}
	f.Set("type", "flex")
	return f
}

/**
 * stretch, start, flex-start, flex-end, end, center, baseline
 */
func (f *Flex) AlignItems(value interface{}) *Flex {
	f.Set("alignItems", value)
	return f
}

/**
 * css 类名
 */
func (f *Flex) ClassName(value interface{}) *Flex {
	f.Set("className", value)
	return f
}

/**
 * 布局方向: column, row
 */
func (f *Flex) Direction(value interface{}) *Flex {
	f.Set("direction", value)
	return f
}

/**
 * 子元素, 数组表示多个
 */
func (f *Flex) Items(value interface{}) *Flex {
	f.Set("items", value)
	return f
}

/**
 * start, flex-start, center, end, flex-end, space-around, space-between, space-evenly
 */
func (f *Flex) Justify(value interface{}) *Flex {
	f.Set("justify", value)
	return f
}

/**
 * 自定义样式
 */
func (f *Flex) Style(value interface{}) *Flex {
	f.Set("style", value)
	return f
}

/**
 * 指定为 flex 渲染器。
 */
func (f *Flex) Type(value interface{}) *Flex {
	f.Set("type", value)
	return f
}

File: origins/Flex.java
package com.wcz0.renderers;

/**
 * Flex
 * @author wcz0
 */
public class Flex extends BaseRenderer {

	Flex() {
		this.set("type", "flex");
	}

	/**
	 * stretch, start, flex-start, flex-end, end, center, baseline
	 */
	public Flex alignItems(Object value) {
		return (Flex) this.set("alignItems", value);
	}

	/**
	 * css 类名
	 */
	public Flex className(Object value) {
		return (Flex) this.set("className", value);
	}

	/**
	 * 布局方向: column, row
	 */
	public Flex direction(Object value) {
		return (Flex) this.set("direction", value);
	}

	/**
	 * 子元素, 数组表示多个
	 */
	public Flex items(Object value) {
		return (Flex) this.set("items", value);
	}

	/**
	 * start, flex-start, center, end, flex-end, space-around, space-between, space-evenly
	 */
	public Flex justify(Object value) {
		return (Flex) this.set("justify", value);
	}

	/**
	 * 自定义样式
	 */
	public Flex style(Object value) {
		return (Flex) this.set("style", value);
	}

	/**
	 * 指定为 flex 渲染器。
	 */
	public Flex type(Object value) {
		return (Flex) this.set("type", value);
	}
}